### DISCUSSION TITLE
All flags automatically enabled on dev environment?

### DISCUSSION THREAD

**Discussion by simonfaltum** [sergiodxa/flagged#473] [Q&A] [answered]
Is there any obvious way I could use FlagsProvider or something similar, so I could always have all flags enabled in my development environment?
My main pain point is that if I am in dev environment, I don't want to have to create all flags in my database or where I store them every time I boot up.
My initial thought was to look for the .env having a variable set to "Development" and check for that.

What do you think about such a feature? 
My own idea would be to make a pull request, updating useFeature hook to look for a "magic" environment variable being set - it would obviously not work with the useFeatures hook.

**Comment by sergiodxa** [comment, marked as answer]
Flagged doesn't have any way to get the flags by itself, you provide them, and they could be set at build time or at runtime, they could even come from the database/an API.

If you know your list of flags you can do the check yourself

```tsx
let isDev = process.env.NODE_ENV !== "production"

return <FlagsProvider features={{ v2: isDev || featureFromAPI}}><App /></FlagsProvider>
```

Using a special env variables will mean the user of the lib needs to setup their build tool to replace `process.env.FLAGGED_MAGIC_ENV` with the value from `process.env` at build time, not all tools do that, Create React App requires a special prefix for env variables, same with Next, Remix doesn't even do that at all.
